### PR TITLE
Fix fs access anti pattern

### LINKED ISSUE #60: Broken since v3.7.1
Hi @tcort,

Thanks for your effort on this awesome tool, I just found that `markdown-link-check` is now broken, in my cases, it just hangs without any output, even with `-v` & `-p` parameter, for 10 mins, 20 mins and longer, I don't know how long it'll hang but seems forever. Builds of [`nodejs/docker-node`](https://travis-ci.com/nodejs/docker-node/builds/80703281) now also failed because there is no any response from it, and it doesn't exit.

Looks like v3.6.2 still works very well, any idea?

Many thanks again!

### PR DESCRIPTION
**Pull request by anchnk**
Attempt to fix #60 

The read stream error handling logic should be discussed, as of now it's a bit rough and it exists if any errors happen. This might not be what users want. Feel free to comment if you  have suggestions on that. Moreover I don't think that PR releases resources the cleanest way possible so that might be necessary to log an issue and spend more time on this. As of now it should fix the project.

### CODE REVIEW

File: markdown-link-check [hunk 1]
```diff
@@ -51,36 +51,40 @@ program
                 parsed.pathname = parsed.pathname.substr(0, parsed.pathname.lastIndexOf('/') + 1);
             }
             opts.baseUrl = url.format(parsed);
-        } catch (err) { /* ignore error */ 
+        } catch (err) { /* ignore error */
```
Comment (PeterDaveHello) [inline markdown-link-check]: This change is not related but a styling.

File: markdown-link-check [hunk 2]
```diff
@@ -51,36 +51,40 @@ program
...
-            } else {
-                opts.baseUrl = 'file://' + path.dirname(path.resolve(filenameOrUrl));
-                stream = fs.createReadStream(filenameOrUrl);
-            }
-          });
+        opts.baseUrl = 'file://' + path.dirname(path.resolve(filenameOrUrl));
+        stream = fs.createReadStream(filenameOrUrl);
     }
-        
+
```
Comment (PeterDaveHello) [inline markdown-link-check]: This change is not related but a styling.

File: markdown-link-check [hunk 3]
```diff
@@ -99,7 +103,7 @@ stream.on('data', function (chunk) {
                     opts.ignorePatterns = config.ignorePatterns;
                     opts.replacementPatterns = config.replacementPatterns;
                     opts.httpHeaders = config.httpHeaders;
-
```
Comment (PeterDaveHello) [inline markdown-link-check]: This change is not related but a styling.

File: markdown-link-check [hunk 4]
```diff
@@ -90,7 +94,7 @@ stream.on('data', function (chunk) {
             if (!err) {
                 let configStream = fs.createReadStream(program.config);
                 let configData = '';
-
```
Comment (PeterDaveHello) [inline markdown-link-check]: This change is not related but a styling.

File: markdown-link-check [hunk 5]
```diff
@@ -51,36 +51,40 @@ program
...
 
 opts.showProgressBar = (program.progress === true); // force true or undefined to be true or false.
 opts.quiet = (program.quiet === true);
 opts.verbose = (program.verbose === true);
 
 let markdown = ''; // collect the markdown data, then process it
-stream.on('data', function (chunk) {
+
+stream
+  .on('data', function (chunk) {
```
Comment (PeterDaveHello) [inline markdown-link-check]: This change is not related but a styling.

File: markdown-link-check (hunk 1 above)
Comment (anchnk) [reply markdown-link-check]: Absolutely it's part of b7e4256 which atomically indicates changes related to styling. 

Does your comments mean I should issue a PR simply for styling ?

**Comment by tcort** [comment]
Thanks for the patch! Published in v3.7.2.

File: markdown-link-check (hunk 1 above)
Comment (PeterDaveHello) [reply markdown-link-check]: No, I mean it's not related to the topic, however I'm not the maintainer/author, just leave my comment without any request ;)

**Pull request merged**